enterprise/suppress-user/suppressUser.go

```diff
@@ -47,29 +49,41 @@ type apiResponse struct {
 }
 
 type sourceRegulation struct {
-	Canceled  bool     `json:"canceled"`
-	UserID    string   `json:"userId"`
-	SourceIDs []string `json:"sourceIds"`
+	Canceled    bool     `json:"canceled"`
+	WorkspaceID string   `json:"workspaceId"`
+	UserID      string   `json:"userId"`
+	SourceIDs   []string `json:"sourceIds"`
 }
 
 func (suppressUser *SuppressRegulationHandler) setup(ctx context.Context) {
+	suppressUser.RegulationBackendURL = configBackendURL
+	switch suppressUser.ID.Type() {
```

Comment:
I would suggest [this](https://github.com/rudderlabs/rudder-server/blob/master/services/controlplane/features/client.go#L93) approach mainly for consistency. I also find it more idiomatic, but that is up for discussion.

Reply:
Usually, it is considered to be a bad practice requiring from the consumer of an exported type to have to know both about the generic interface type and the concrete implementation types, since it beats the original purpose of the interface, that is abstracting from the underlying implementation(s).